libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -238,64 +244,164 @@ string IRGenerator::generateInitialAssignment(VariableDeclaration const& _varDec
 	return generator.code();
 }
 
-string IRGenerator::constructorCode(ContractDefinition const& _contract)
+pair<string, map<ContractDefinition const*, string>> IRGenerator::evaluateConstructorArguments(
+	ContractDefinition const& _contract
+)
 {
-	// Initialization of state variables in base-to-derived order.
-	solAssert(!_contract.isLibrary(), "Tried to initialize state variables of library.");
-
-	using boost::adaptors::reverse;
-
-	ostringstream out;
+	IRGeneratorForStatements generator{m_context, m_utils};
+	auto listParamNames = [&](
+		FunctionDefinition const* baseConstructor,
+		vector<ASTPointer<Expression>> const* arguments
+	) -> string
+	{
+		string params;
+		for (size_t i = 0; i < arguments->size(); ++i)
+			params += (params.empty() ? "" : ", ") + generator.evaluateExpression(
```

Comment:
How do you know `evaluateExpression` only returns a single yul expression? I think you should assign each argument to its own new local variable and then return the tuple of these variables.

Reply:
Evaluate expression returns one IRVariable, which is one parameter. Isn't it enough?

Reply:
if there are multiple expressions to get that one variable, they are returned from this function as yul source

Reply:
Ah you are right, the `.code()` is extracted at the end!